### PR TITLE
Tn image index

### LINKED ISSUE #2: Image Index
#### As a user I want the homepage to display all saved images.

__Story__:

Now that we are saving images for our users, we want them to be able to view
the list of images in the system. You might be thinking, "what if my users only
want to view images of cats", or "what if my users only want to see the images
they uploaded". Those are great questions. Maybe you will have the opportunity
to add those features in a future story. For now, we will keep it simple and
quickly deliver a little more value to our users.

__Acceptance criteria__:
- [x] New images that are added show up on the homepage.
- [x] These images are persisted if the browser is closed or even if the
  server is restarted.
- [x] Images are not displayed wider than 400px.
- [x] Newest images appear first.

__Dependencies__:
- Save Image Link

### PR DESCRIPTION
**Pull request by justinpearsonappfolio**
Closes #2

### CODE REVIEW

File: test/controllers/images_controller_test.rb [hunk 1]
```diff
@@ -37,4 +37,34 @@ def test_show
...
     assert_select '.js-show-image', 1
   end
+
+  def test_index
+    image_urls_from_db = []
+    10.times do |i|
+      image_url = "http://app.com/#{i}.jpg"
+      image_urls_from_db.append image_url
+      image = Image.new(url: image_url)
+      image.save!
```
Comment (connorproctor) [inline test/controllers/images_controller_test.rb]: Minor improvement suggestion: `.new` and `.save!` can be combined into a `.create!`.

File: test/controllers/images_controller_test.rb [hunk 2]
```diff
@@ -37,4 +37,34 @@ def test_show
...
+    end
+    image_urls_from_db.reverse!
+
+    get root_path
+
+    assert_response :ok
+    assert_select 'img.index-images', Image.count
+
+    assert_select 'img.index-images' do |image_elements|
+      image_elements.zip(image_urls_from_db).each do |element, url|
```
Comment (connorproctor) [inline test/controllers/images_controller_test.rb]: Nice use of zip!

File: test/controllers/images_controller_test.rb [hunk 3]
```diff
@@ -37,4 +37,34 @@ def test_show
...
+        assert_equal element[:src], url
+      end
+    end
+  end
```
Comment (connorproctor) [inline test/controllers/images_controller_test.rb]: Does this accurately test that images are being ordered by the `created_at` timestamp?

File: test/controllers/images_controller_test.rb (hunk 1 above)
Comment (lcguang) [reply test/controllers/images_controller_test.rb]: Got it! Thanks!

File: test/controllers/images_controller_test.rb (hunk 3 above)
Comment (lcguang) [reply test/controllers/images_controller_test.rb]: We changed it to ordering images from the database by 'created_at'.

File: test/controllers/images_controller_test.rb [hunk 4]
```diff
@@ -37,4 +37,31 @@ def test_show
     assert_select '.js-show-image-header', 'Show Image'
     assert_select '.js-show-image', 1
   end
+
+  def test_index
+    10.times do |i|
+      image_url = "http://app.com/#{i}.jpg"
+      Image.create!(url: image_url)
```
Comment (connorproctor) [inline test/controllers/images_controller_test.rb]: Will these 10 images have different `created_at` timestamps?

**Pull request merged**